src/extension/app-schema/app-schema-web/src/main/java/org/geoserver/complex/ComplexToSimpleFeatureCollection.java

```diff
@@ -0,0 +1,216 @@
+        return new ComplexToSimpleFeatureIterator(delegate.features());
+    }
+
+    @Override
+    public SimpleFeatureCollection subCollection(Filter filter) {
+        return null;
+    }
+
+    @Override
+    public SimpleFeatureCollection sort(SortBy order) {
+        return null;
+    }
+
+    @Override
+    public String getID() {
+        return delegate.getID();
+    }
+
+    @Override
+    public void accepts(FeatureVisitor visitor, ProgressListener progress) throws IOException {
+        DataUtilities.visit(this, visitor, progress);
+    }
+
+    @Override
+    public ReferencedEnvelope getBounds() {
+        return delegate.getBounds();
+    }
+
+    @Override
+    public boolean contains(Object o) {
+        return delegate.contains(o);
+    }
+
+    @Override
+    public boolean containsAll(Collection<?> o) {
+        return delegate.containsAll(o);
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return delegate.isEmpty();
+    }
+
+    @Override
+    public int size() {
+        return delegate.size();
+    }
+
+    @Override
+    public Object[] toArray() {
+        return getFeaturesList().toArray();
+    }
+
+    @Override
+    public <O> O[] toArray(O[] a) {
+        return getFeaturesList().toArray(a);
+    }
+
+    private List<SimpleFeature> getFeaturesList() {
+        try (SimpleFeatureIterator featureIterator = features()) {
+            List<SimpleFeature> featuresList = new ArrayList<>();
+            while (featureIterator.hasNext()) {
+                featuresList.add(featureIterator.next());
+            }
+            return featuresList;
+        }
+    }
+
+    /** Feature iterator wrapper, converts every complex feature into a simple feature. */
+    class ComplexToSimpleFeatureIterator implements SimpleFeatureIterator {
+
+        private FeatureIterator<Feature> delegate;
+
+        public ComplexToSimpleFeatureIterator(FeatureIterator<Feature> delegate) {
+            this.delegate = delegate;
+        }
+
+        public boolean hasNext() {
+            return delegate.hasNext();
+        }
+
+        public SimpleFeature next() throws NoSuchElementException {
+            SimpleFeature simpleFeature = convert(delegate.next());
+            LOG.finer(() -> "Converted simple feature: " + simpleFeature);
+            return simpleFeature;
+        }
+
+        /**
+         * Transform the original complex feature into a simple feature, using the convention and
+         * rules. Returns the resulting simple feature.
+         */
+        private SimpleFeature convert(Feature feature) {
+            SimpleFeatureBuilder builder = new SimpleFeatureBuilder(featureType);
+            for (PropertyDescriptor descriptor : featureType.getDescriptors()) {
+                String simpleName = descriptor.getName().getLocalPart();
+                Object attributeValue = getComplexAttributeValue(simpleName, feature);
+                builder.set(descriptor.getName(), attributeValue);
```

Comment:
So here the full, qualified name is used, however, in getComplexAttributeValue, the lookup is made based on the simple name. Confusing?